### DISCUSSION TITLE
Failing to resolve git dependency in a Flutter build

### DISCUSSION THREAD

**Discussion by miguelcmedeiros** [codemagic-ci-cd/codemagic-docs#1724] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

The app that it's failing to build is within a mono repo project, where we use melos to solve and keep our dependencies up-to-date. Not sure if this is relevant for the issue, but thought it was worth mentioning.
During the step where we bootstrap our project: `dart run melos bootstrap`, it fails with the following error:
```
[superlist]: Got dependencies!
  ✓ superlist
    └> packages/superlist
  - super_design
    └> packages/super_design
       └> Failed to install.

Resolving dependencies...
Because super_design depends on super_text_layout from git which doesn't exist (Could not find git ref '2cc679bdc1677554bf9249e1578351aa54a0e000' (fatal: bad object 2cc679bdc1677554bf9249e1578351aa54a0e000)), version solving failed.
BootstrapException: Failed to install.: super_design at /Users/builder/clone/packages/super_design.


Build failed :|
Step 4 script `Fetch flutter dependencies` exited with status code 1
```

The dependency mentioned above (`super_text_layout`) is part of another public repo ([`super_editor`](https://github.com/superlistapp/super_editor)), but somehow it fails to resolve the git ref.

### Expected results

Dependencies are resolved successfully

### Actual results

Fails with:
```
[superlist]: Got dependencies!
  ✓ superlist
    └> packages/superlist
  - super_design
    └> packages/super_design
       └> Failed to install.

Resolving dependencies...
Because super_design depends on super_text_layout from git which doesn't exist (Could not find git ref '2cc679bdc1677554bf9249e1578351aa54a0e000' (fatal: bad object 2cc679bdc1677554bf9249e1578351aa54a0e000)), version solving failed.
BootstrapException: Failed to install.: super_design at /Users/builder/clone/packages/super_design.


Build failed :|
Step 4 script `Fetch flutter dependencies` exited with status code 1
```

### Build id (optional)

6475cb4cc6270280aace5dd5

**Comment by icarusdust** [comment]
Hi @miguelcmedeiros, first off, please note that as your team is on a paid plan, you should have access to the chat widget in the UI which connects you with our support team for quicker response. Please use that widget for your future queries as it is a pad feature.

As for the issue, try to upgrade the `super_editor `version, looks like it is failing with version inconsistency.

**Reply by icarusdust** [reply, marked as answer]
Actually, i see that you have already resolved the issue.
